src/main/scala/mesosphere/marathon/core/readiness/impl/SprayReadinessCheckExecutor.scala

```diff
@@ -0,0 +1,81 @@
+package mesosphere.marathon.core.readiness.impl
+
+import akka.actor.ActorSystem
+import akka.util.Timeout
+import mesosphere.marathon.core.readiness.ReadinessCheckExecutor.ReadinessCheckSpec
+import mesosphere.marathon.core.readiness.{ HttpResponse, ReadinessCheckExecutor, ReadinessCheckResult }
+import org.slf4j.LoggerFactory
+import rx.lang.scala.Observable
+import spray.client.pipelining._
+import spray.http.HttpHeaders.`Content-Type`
+import spray.http.{ HttpResponse => SprayHttpResponse, _ }
+
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+import scala.util.Success
+import scala.util.control.NonFatal
+
+/**
+  * A Spray-based implementation of a ReadinessCheckExecutor.
+  */
+private[readiness] class SprayReadinessCheckExecutor(support: SprayReadinessCheckExecutor.Support)
+    extends ReadinessCheckExecutor {
+
+  import scala.concurrent.ExecutionContext.Implicits.global
+
+  override def execute(readinessCheckSpec: ReadinessCheckSpec): Observable[ReadinessCheckResult] = {
+    support
+      .intervalObservable(readinessCheckSpec.interval)
+      .flatMap(_ => Observable.from(support.executeSingleCheck(readinessCheckSpec)))
+      .takeUntil(_.ready)
+  }
+}
+
+private[readiness] object SprayReadinessCheckExecutor {
+
+  /** Support class for better testability. */
+  class Support(actorSystem: ActorSystem) {
+    import scala.concurrent.ExecutionContext.Implicits.global
+
+    private[this] val log = LoggerFactory.getLogger(getClass)
+    private[this] implicit val system = actorSystem
```

Comment:
Testability: I though that I would end up mocking some of the methods. Now that I didn't do that, I can inline them in the other class again.

`actorSystem` parameter, fine. Then we have implicitly two parameter lists but why not.